### PR TITLE
template: add template_destpath variable to templates

### PR DESCRIPTION
**Pull request by virtuald**
##### SUMMARY
Add template_destpath variable to templates

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
template action plugin

### CODE REVIEW

**Comment by virtuald** [comment]
I fixed the PEP8 issue, the build seems to have failed for a non-build related reason.

**Comment by gundalow** [comment]
I've hit rebuild on the Shippable job, looks like it may have been hit by the GitHub issues today

File: lib/ansible/modules/files/template.py [hunk 1]
```diff
@@ -30,13 +30,14 @@
        (U(http://jinja.pocoo.org/docs/)) - documentation on the template
        formatting can be found in the Template Designer Documentation
        (U(http://jinja.pocoo.org/docs/templates/)).
-     - "Six additional variables can be used in templates:
+     - "Seven additional variables can be used in templates:
        C(ansible_managed) (configurable via the C(defaults) section of C(ansible.cfg)) contains a string which can be used to
```
Comment (gundalow) [inline lib/ansible/modules/files/template.py:34]: I wonder if this was written as
```
- "Additional variables can be used in templates:
- C(ansible_managed) (configurable via the C(defaults) section...
- C(template_host) contains the...
```
would result in a more readable http://docs.ansible.com/ansible/template_module.html

Comment (virtuald) [reply lib/ansible/modules/files/template.py:34]: Perhaps, though I think that's out of scope for this particular PR.

**Comment by virtuald** [comment]
The test failure isn't my bug:

```
2017-05-31 15:41:47 ERROR: Found 2 pylint issue(s) which need to be resolved:
2017-05-31 15:41:47 ERROR: lib/ansible/plugins/callback/profile_tasks.py:141:23: too-few-format-args Not enough arguments for format string (0%)
2017-05-31 15:41:47 ERROR: lib/ansible/plugins/callback/profile_tasks.py:141:62: too-many-format-args Too many arguments for format string (0%)
```

File: lib/ansible/modules/files/template.py (hunk 1 above)
Comment (dagwieers) [inline lib/ansible/modules/files/template.py]: I wouldn't keep a count of how many additional variables can be used. Just list them !

Comment (dagwieers) [reply lib/ansible/modules/files/template.py:34]: Never waste a good opportunity for improving this too :-)

**Comment by dagwieers** [comment]
-label windows

**Comment by jctanner** [comment]
bot_skip

**Comment by Akasurde** [comment]
@virtuald Are you still working on this ? Thanks

needs_info

**Comment by virtuald** [comment]
I'm not working on any projects that use Ansible at this time.

**Comment by Akasurde** [comment]
superseded by https://github.com/ansible/ansible/pull/52015

**Pull request closed without merging**